src/api/middleware/validate_username.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response, NextFunction } from "express";
+import expressAsyncHandler from "express-async-handler";
+import user from "../dao/user";
+
+const USERNAME_TAKEN: number = 409;
+const MISSING_CREDENTIAL: number = 401;
+
+export const validate_username = expressAsyncHandler(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const username = req.body.username;
+
+    if (!username) {
+      res.status(MISSING_CREDENTIAL).send({
+        status: MISSING_CREDENTIAL,
+        message: "Missing username",
+      });
+      return
+    }
+
+    if (!req.body.password) {
+      res
+        .status(MISSING_CREDENTIAL)
+        .send({ status: MISSING_CREDENTIAL, message: "Missing password" });
+      return
+    }
+
+    if (!req.body.roleId) {
+      res
+        .status(MISSING_CREDENTIAL)
+        .send({ status: MISSING_CREDENTIAL, message: "Missing roleId" });
+      return
+    }
+
+    let name_candidate = req.body.username;
```

Comment:
potentially redundant variable that's already defined on line 10? If so remove

Reply:
Resolved in 

https://github.com/disney-parent-pass/back-end/pull/8/commits/e3847fde9f149cf3fdde0f77564e3c855f49da50